### ISSUE TITLE
Can we have the Frame.Navigate(object content, object extraData) override?

### ISSUE THREAD

**Issue by luojunyuan** [Kinnara/ModernWpf#329]
Wanna this because I can hold instances of each page in a single window.
when I use `Frame.Navigate()` I can only use `Uri` (#89) or `Type` to navigate, both of which would recreate a Page instance. 
If we can have `public bool Navigate(object content, object extraData, NavigationTransitionInfo infoOverride)` like the original Frame. I could then do
```csharp
AboutPage aboutPage = new AboutPage();

MyFrame.Navigate(aboutPage, null, transitionInfo);
```

**Comment by Lunchb0ne** [comment]
Yes, this would be really helpful, I'm new to WPF and I can't find a proper way to do this too

**Comment by Kinnara** [comment]
Implemented on the active 1.x line by #705.

`ModernWpf.Controls.Frame` now exposes both instance-navigation transition overloads:

```csharp
bool Navigate(object content, NavigationTransitionInfo infoOverride)
bool Navigate(object content, object extraData, NavigationTransitionInfo infoOverride)
```

The requested usage works directly:

```csharp
var aboutPage = new AboutPage();
MyFrame.Navigate(aboutPage, null, transitionInfo);
```

The explicit object remains caller-owned rather than entering the type-based page cache. Runtime coverage verifies that the exact page instance becomes `Frame.Content`, `extraData` reaches the WPF navigation event, the transition override is applied, and transient override state is cleared afterward.

Current `FrameApiTests`: 3 passed, 0 failed, 0 skipped. Closing as completed.
